2.13/tests/test/test.js

```diff
@@ -675,3 +747,71 @@ function validateJsonSchema(body, schema) {
     }
     return '';
 }
+
+function validateCatalogSchema(tempBody, schema_type, action) {
+    preparedRequest()
+        .get('/v2/catalog')
+        .set('X-Broker-API-Version', apiVersion)
+        .auth(config.user, config.password)
+        .expect(200)
+        .expect('Content-Type', /json/)
+        .end(function(err, res){
+            if (err) {
+                var message = "Get catalog result failed: " + err;
+                return message;
+            }
+            var catalog = JSON.parse(JSON.stringify(res.body));
+            var schemaCheckResults = parametersSchemaCheck(catalog, tempBody.service_id, tempBody.plan_id, schema_type, action, tempBody.parameters);
+            if (schemaCheckResults != '') {
+                var message = "Validate schema parameters failed!";
+                return message;
+            }
+        })
+    return '';
+}
+
+function containsKeyValue(obj, key, value) {
+    if (!obj) {
+        return null;
+    }
+    if (obj[key] === value) {
+       return obj;
+    }
+    if (Array.isArray(obj)) {
+        for (var i in obj) {
+            var found = containsKeyValue(obj[i], key, value);
+            if (found) return found;
+        }
+    } else if (typeof obj == "object") {
+        for (var p in obj) {
+            if (p === key && obj[p] === value) {
+                return obj;
+            }
+            var found = containsKeyValue(p, key, value);
+            if(found) return found;
+        }
+    }
+    return null;
+}
+
+function parametersSchemaCheck(catalog, service_id, plan_id, schema_type, action, parameters){

```

Comment:
I think all variables should use camel case, and we can use underscore for fields in a object.